File: Nature of Code/Pygame/Unit 0/Random Walker 1.0/walker.py
"""Clase que crea un objeto Walker"""

from random import randint, random
import pygame

class Walker:
    """Creación de objetos tipo Walker"""
    def __init__(self, x: int, y: int):
        self.x = x
        self.y = y
    def walk(self):
        """El objeto se mueve a una dirección aleatoria"""

        #Caminar en 4 direcciones
        # step_dir = randint(0, 3)
        # if step_dir == 0:
        #     self.x += 1
        # elif step_dir == 1:
        #     self.x -= 1
        # elif step_dir == 2:
        #     self.y += 1
        # elif step_dir == 3:
        #     self.y -= 1
        #Caminar en 9 direcciones
        xstep: int = randint(-1, 1)
        ystep: int = randint(-1, 1)
        self.x += xstep
        self.y += ystep                  
    def show(self, window):
        """Muestra el objeto en pantalla """
        pygame.draw.circle(window, (0, 0, 0, 100), (self.x, self.y), 2)

File: Nature of Code/PJ5/Unit 0 - Randomness/Random Walker 1.0/walker.js
class Walker{
    constructor(x, y)
    {
        this.x = x;
        this.y = y;
    }
    walk()
    {
        //Caminar en 4 direcciones
        /* let stepDir = floor(random(0, 4));
        switch(stepDir)
        {
            case 0:
                this.x += 1;
            break;
            case 1:
                this.x -= 1;
            break;
            case 2:
                this.y += 1;
            break;
            case 3:
                this.y -= 1;
            break;    
        } */

        //Caminar en 9 direcciones, X y Y pueden moverse -1, 0, 1
        let xstep = floor(random(3)) - 1;
        let ystep = floor(random(3)) - 1;

        //Caminar en cualquier dirección pero en distancias decimales entre -1 y 1
        /* 
        let xstep = random(-1, 1);
        let ystep = random(-1, 1); 
        */
        
        this.x += xstep;
        this.y += ystep;
    }
    show()
    {
        stroke(255)      
        point(this.x, this.y);
    }
}